internal/pkg/template/permissions_boundary_integration_test.go

```diff
@@ -0,0 +1,31 @@
+//go:build integration || localintegration
+
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package template
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestPermissions_Boundary(t *testing.T) {
+	t.Run("every CloudFormation template must contain conditional permissions boundary field for all IAM roles", func(t *testing.T) {
+		err := filepath.WalkDir("templates", func(path string, di fs.DirEntry, err error) error {
+			contents, _ := os.ReadFile(path)
```

Comment:
nit:
```suggestion
			contents, err := os.ReadFile(path)
			require.NoError(err, "read file at %s", path)
```

Reply:
Ah, I had taken that `err` out because `ReadFile` was returning errors when trying to read directories, like `read templates: is a directory`

Reply:
added `IsDir()`